app/actions/update_route_destinations.rb

```diff
@@ -126,6 +129,23 @@ def update_processes(processes_to_ports_map)
         raise Error.new(e.message)
       end
 
+      def compare_destination_hashes(new_route_mappings, existing_route_mappings)
+        new_route_mappings.reject do |new|
+          matching_route_mapping = existing_route_mappings.find do |existing|
+            new[:app_guid] == existing[:app_guid] &&
+              new[:process_type] == existing[:process_type] &&
+              new[:route_guid] == existing[:route_guid] &&
+              new[:app_port] == existing[:app_port]
```

Comment:
if you like, can one or two line this:

```
new.slice(:app_guid, :process_type, :route_guid, :app_port) == existing.slice(:app_guid, :process_type, :route_guid, :app_port)
```

or

```
attrs = [:app_guid, :process_type, :route_guid, :app_port]
new.slice(attrs) == existing.slice(attrs)
```

Reply:
or maybe use https://apidock.com/rails/v5.2.3/Hash/except if we want to compare everything but `protocol`?  that might clearer in intent